search/search_repository.go

```diff
@@ -685,6 +697,31 @@ func (r *GormSearchRepository) listItemsFromDB(ctx context.Context, criteria cri
 		return nil, 0, errors.NewBadParameterError("expression", criteria)
 	}
 
+	if childIterations {
+		itrRepo := iteration.NewIterationRepository(r.db)
+
+		allitrs := []string{}
+		for _, itr := range iterations {
+			childItrs, err := itrRepo.LoadChildren(ctx, itr)
+			if err != nil {
+				// Since this is a search, missing iteration is not really an error.
+				if ok, _ := errors.IsNotFoundError(err); ok {
+					continue
+				}
+				return nil, 0, errors.NewInternalError(ctx, errs.Wrap(err, "failed to load child iteration"))
+
+			}
+			allitrs = append(allitrs, fmt.Sprintf("Fields->>'system.iteration' = '%s'", itr))
```

Comment:
@baijum can you please prefix the `fields` column with the name of the work item's table? 

```go
allitrs = append(allitrs,  fmt.Sprintf(
  "%s.'%s' = %s",
  workitem.Column(WorkItemStorage{}.TableName(), "fields"),
  workitem.SystemIteration, 
  itr,
)
```

I did just break things in lines for reading purposes. You don't need to do that.